### ISSUE TITLE
Inventory from another multistore group affects product in the FO of another group

### ISSUE THREAD

**Issue by ocio87** [PrestaShop/PrestaShop#28697]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

It is identified that in a product when it has two attributes (Size and different colors), in the FrontOffice it shows the color although there is no existence in that group but in another group that does not share units.

https://drive.google.com/file/d/1drMyp7vQXJaVcQL86aqYv1q3PcMRqRz7/view

### Steps to reproduce

1. Go to BO > catalog > products > choose or add a product with multiple combinations (size + colors for example)
2. On product detail page > go to combinations tab > add or edit combination with size and colors attributes > generate them
3. Check all those combinations > add the same quantity for all
4. Go to FO >  click on the product concerned > check if all combinations are in stock
5. Go to BO > shop parameters > general > enable multistore
6. Go to advanced parameters > edit the default group > set to yes "Share available quantities to sell" switch button
7. Add a new shop group > add a name > save it
8. Add a new shop > add a name and choose the new shop group created > save it
9. Click on "Click here to set a URL for this shop." link to set an url for this new shop > set this url on "virtual url field" > save it
10. Go to shop parameters > product settings page > product page block > set to no "Display unavailable product attributes on the product page" switch button > save it
11. Stay on this page > go to product stock block > set with content the "Label of in-stock products" field
12. Go to catalog > products > choose the product with combinations
13. Choose the default shop and set a quantity to 0 of a combination > click on save button and click on preview  button
14. Go to the new tab on FO preview page of the product > choose the combination which is set to 0 
15. See error, the "in stock" label is still displayed, and it's impossible to choose the concerned combination. we go back on the first combination automatically

### PrestaShop version(s) where the bug happened

1.7.8.5

### PHP version(s) where the bug happened

7.4

**Comment by hibatallahAouadni** [comment]
Hello @ocio87 

Could you please specify what do you mean by "group" in this sentence:
>  in the FrontOffice it shows the color although there is no existence in that group but in another group that does not share units.

Do you mean a group shop? and do you mean by "inventory" stock?
As far as I understand, you encounter a problem when you share the products between the shop's groups and not sharing the stock, is that so? 
I'm really sorry but your steps to reproduce is a little bit confusing and I didn't understand well the issue :disappointed: 

Waiting for your feedback.
Thanks!

**Comment by ocio87** [comment]
Hello, there are two groups of stores configured:

In group one with name Default there are 3 stores which share stock between them.

In group 2, it has the name Premium Plaza, which only has one store configured and no stock is shared.

![image](https://user-images.githubusercontent.com/12537826/172687279-a8c194f6-6341-4224-a8f1-3a7ddd5008ce.png)


If a product with size attribute and different colors does not have stock in size 36 red color in group 1 of store 1 but if there is stock in that size 36 and red color in group 2 of store 4, it is enabling the color in group 1, however it is not allowed to select in group 1.

It is not the expected behavior because they do not share stock between the groups.

**Comment by sLorenzini** [comment]
Hello @ocio87,

I coudn't reproduce your issue.

My configuration:

2 groups of shop:
- Default with 2 shops (PS 1785 & default 2) and "Share available quantities for sale" switch set to **yes**
- attributes with 1 shop (shop 1) and "Share available quantities for sale" switch set to **no**

1 product on each shop with S/M/L sizes and Green/Yellow/Red colors as attributes.

On shop 1 (group 2), the quantity is set to 100. On FO, this combination is in stock.

On PS 1785 & Default 2 (group 1) , the quantity is set to 0 with L and Red colors combination, on each shop (thanks to swtich button set to yes in multistore parameter). On FO, this combination is out of stock and not in stock with the parameter on group 2.

Please check below the screen record:

https://user-images.githubusercontent.com/2854834/172824854-aca111b1-eba5-47d9-9589-73a3ddc92f9d.mp4

I've tested also with:

- "Share available quantities for sale" switch set to no on any group => good, stock is different in FO on each store
- "Share available quantities for sale" switch set to yes on default group => good, stock is different in FO on each store
- "Share available quantities for sale" switch set to yes on 2nd group  => good, stock is different in FO on each store
- "Share available quantities for sale" switch set to yes on all group =>  good, stock is different in FO on each store

Could you provide us more details please? or maybe I forgot something to be able to reproduce your issue 😅 .

Waiting for your feedback
Thanks

**Comment by ocio87** [comment]
Hello,

Sorry, I missed a configuration that is involved in the bug.

In group 1 the stock is fine. The novelty is when stock is increased in group 2 and this increase is shown in group 1 although it is not allowed to be selected when the configuration is that it does not show the attribute when there is no stock.

![image](https://user-images.githubusercontent.com/12537826/172843964-ace52333-5266-4f7f-aa45-6adc3f46f2bc.png)

**Comment by sLorenzini** [comment]
hello @ocio87,

Indeed with this option set to "No", I reproduce your issue with ps1785 / ps1778. I'll add this to the backlog so it can be fixed.

See screen record below:

https://user-images.githubusercontent.com/2854834/172853683-c125d0ad-90a6-41d9-80fc-9e641126c160.mp4

I'll modify a  your steps to reproduce, can you notify me if this is not good steps?

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by ocio87** [comment]
Hello,

That is exactly the reported error.

Thank you very much,

**Comment by ocio87** [comment]
Hello, is there any progress on this bug reported in 2022?

**Comment by ocio87** [comment]
It displays attributes that are not in stock in the store.

To reproduce this error, proceed as follows.

The Meadow color is available in 30 oz and 40 oz capacities and is out of stock in all stores.

![Image](https://github.com/user-attachments/assets/9ae288ff-aacf-4d85-98c8-ae5718e8e928)

I assign 10 units to the 40 oz capacity without assigning units to the 30 oz capacity in the store with shop_id 1, which shares inventory with stores with shop_ids 2 and 3:

![Image](https://github.com/user-attachments/assets/3316ad5c-3e07-439c-907c-398e968b2b5e)

As we can see, the color was only enabled correctly in the 40 oz capacity:

![Image](https://github.com/user-attachments/assets/e16cf633-6789-429d-ad05-8a3a694ebe42)

![Image](https://github.com/user-attachments/assets/edebca86-2aff-4334-93ae-16876194eb61)

If I increase stock in the store with shop_id=5 called Colina in the 30 oz capacity,

![Image](https://github.com/user-attachments/assets/6436015c-0271-432a-be7d-e20811095bf1)

The color is displayed in Charme in shop_id=3, which does not share stock with shop_id=5. And clicking on that color doesn't perform any action:

![Image](https://github.com/user-attachments/assets/6706dbe5-8e35-4b77-aceb-877ef52d2b5c)

If I delete the 30 oz and 40 oz stock in id_shop=1, which shares stock with id_shop=3, and there is stock in those capacities in id_shop=5, it doesn't show the color in id_shop=3, which is correct.

In conclusion, if there is no stock in any of the capacities of that color in the store with id_shop=3, it doesn't show it even if there is stock in id_shop=5, which is Colina. However, if there is stock in the 40 oz capacity and no stock in the 30 oz capacity of that color in the store with id_shop=3, but there is stock in the store with id_shop=5 in the 30 oz capacity, this affects id_shop=3.

**Comment by boo-code** [comment]
Still reproduces on 9.2.x, and @ocio87's 2025-06-05 description narrows it to one query.

The distinction in that comment is the key one: the **colour** is enabled and disabled correctly, and the **capacity** under it is not. Those two lists are built by different code. The colour list and `attributes_quantity` come from `Product::getAttributesGroups()`, which joins stock through `Product::sqlStock()` and therefore through `StockAvailable::addSqlShopRestriction()`. The second and later groups are then narrowed again by an inline query in `ProductController`, and that one has no shop restriction at all:

```php
$id_attributes = Db::getInstance()->executeS('SELECT pac2.`id_attribute` FROM `' . _DB_PREFIX_ . 'product_attribute_combination` pac2' .
    ((!Product::isAvailableWhenOutOfStock(...) && false === (bool) Configuration::get('PS_DISP_UNAVAILABLE_ATTR')) ?
    ' INNER JOIN `' . _DB_PREFIX_ . 'stock_available` pa ON pa.id_product_attribute = pac2.id_product_attribute
    WHERE pa.quantity > 0 AND ' :
    ' WHERE ') . ...
```

Every shop's stock rows live in that one table, so `pa.quantity > 0` is true if **any** shop has stock for the combination — including a shop in a group this one shares nothing with. The attribute survives the wash, is rendered, and does nothing when clicked, because the combination genuinely has no stock here. That is the "it displays the color and clicking it doesn't perform any action" in the report.

Measured on demo product 1 (Size x Colour). `Size:M | Colour:White` set to quantity `0` for shop 1, and a `stock_available` row with `50` added for shop 2, outside shop 1's group. Running that query verbatim with `White` already selected:

```
sizes still offered, query as it is    L, M, S, XL      <- M comes from shop 2
sizes still offered, shop-restricted   L, S, XL
```

**Prepared** against `9.2.x`: `StockAvailable::addSqlShopRestriction(null, null, 'pa')` inserted into that `WHERE`. It emits `AND pa.id_shop = 1 AND pa.id_shop_group = 0` for a shop that does not share stock, and the `id_shop_group = <group> AND id_shop = 0` form for one that does, so both multistore stock modes are covered by the helper the rest of the page already uses.

Two things worth stating plainly. The branch only runs with `Display unavailable product attributes` set to No, which is why @sLorenzini could not reproduce it until @ocio87 mentioned that setting in 2022. And it carries no automated test: the query is inline in a legacy front controller, which no harness in the repository drives — the multishop behat scenarios cover the domain layer, not `ProductController`. The measurement above is the evidence.

The branch is ready; a PR can be opened on request.
